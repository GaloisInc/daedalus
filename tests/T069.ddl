def Main =
  block
    SetStream (arrayStream [1,1,0,1,0])
    one = Example
    SetStream (arrayStream [0,1,0,1,0])
    two = Example

def Example =
  block
    let ?be = UInt8 == 0
    x = Word16
    y = Word16


def Word16 =
  block
    let b1 = UInt8
    let b2 = UInt8
    if ?be then b1 # b2 else b2 # b1

