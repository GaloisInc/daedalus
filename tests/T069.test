T069.ddl
--run
