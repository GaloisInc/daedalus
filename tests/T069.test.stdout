--- Found 1 results:
{ one: { x: 1[16]
       , y: 1[16]
       }
, two: { x: 256[16]
       , y: 256[16]
       }
}
